File: 순위.js
function solution(n, results) {
  var answer = 0;
  const graph = Array.from({ length: n + 1 }, () => Array(n + 1).fill(false));
  for (const [win, lose] of results) {
    graph[win][lose] = true;
  }

  for (let k = 1; k <= n; k++) {
    for (let i = 1; i <= n; i++) {
      for (let j = 1; j <= n; j++) {
        if (graph[i][k] && graph[k][j]) {
          graph[i][j] = true;
        }
      }
    }
  }
  for (let i = 1; i <= n; i++) {
    let count = 0;
    for (let j = 1; j <= n; j++) {
      if (graph[i][j] || graph[j][i]) {
        count++;
      }
    }
    if (count === n - 1) {
      answer++;
    }
  }
  return answer;
}

File: 순위.py
def solution(n, results):
    wins = [set() for _ in range(n + 1)]   
    loses = [set() for _ in range(n + 1)]  

   
    for win, lose in results:
        wins[win].add(lose)
        loses[lose].add(win)

   
    for i in range(1, n + 1):  
        for loser in wins[i]:
            loses[loser].update(loses[i])
        for winner in loses[i]:
            wins[winner].update(wins[i])

    answer = 0
    
    for i in range(1, n + 1):
        if len(wins[i]) + len(loses[i]) == n - 1:
            answer += 1

    return answer